### DISCUSSION TITLE
Displaying every epoch loss during training

### DISCUSSION THREAD

**Discussion by jrfackler** [timeseriesAI/tsai#588] [Q&A] [answered]
I'm using learn.fit_one_cycle to train a model with a large number of epochs and a notebook with Paperspace Gardient.

Displaying the loss after every epoch causes my browser (Chrome) to crash. The code stills runs fine in the background but I'm not able to refresh the browser and see how the training is progressing or how close it is to finishing.

Is there a way to disable displaying the loss after each epoch?

Thank you!

![Image 2](https://user-images.githubusercontent.com/94877460/194584773-d74babdc-3edf-4aa2-a5dd-8463e53665af.jpg)

**Comment by oguiza** [comment]
Hi @jrfackler,
Yes, there's a way to achieve that in fastai (this is not tsai specific). I hope it fixes your issue. You can use ContextManagers. There are some examples in the [Prediction Dynamics](https://github.com/timeseriesAI/tsai/blob/main/tutorial_nbs/09_PredictionDynamics.ipynb) tutorial notebook.
Here's an example of how you can use them:
```python
X, y, splits = get_UCR_data('LSST', split_data=False)
tfms=[None, TSClassification()]
batch_tfms = [TSStandardize(by_sample=True)]
dls = get_ts_dls(X,y,splits=splits,tfms=tfms, batch_tfms=batch_tfms)
learn = ts_learner(dls, InceptionTimePlus, metrics=[accuracy, BalancedAccuracy()], cbs=[ShowGraph(), PredictionDynamics()])
with ContextManagers([learn.no_logging(), learn.no_bar()]): 
    learn.fit_one_cycle(50, 1e-3)
learn.plot_metrics()
```
learn.no_logging() avoids displaying the loss. learn.no_bar() avoids displaying the progress bar.

**Comment by jrfackler** [comment, marked as answer]
Thank you very much!!! That worked perfectly.
